src/status_im/ui/screens/wallet/events.cljs

```diff
@@ -18,107 +19,114 @@
          (on-error err))))
     (on-error "web3 or account-id not available")))
 
+(defn assoc-error-message [db error-type err]
+  (assoc-in db [:wallet :errors error-type] (or (when err (str err))
+                                                :unknown-error)))
+
+(defn clear-error-message [db error-type]
+  (update-in db [:wallet :errors] dissoc error-type))
```

Comment:
Maybe make a note/ticket to figure out some generic mechanism.

Reply:
ok I will make a note, we could indeed have an error key that follows such pattern for all possible error that need to be fed back to the user